lib/replicator.js

```diff
@@ -84,6 +86,15 @@ class Attachable {
       this._detach(this.refs[this.refs.length - 1]).reject(err)
     }
   }
+
+  setTimeout (r, ms) {
+    if (typeof ms !== 'number') return
+    if (r.timeout !== null) return
+
+    r.timeout = setTimeout(() => {
+      if (r.context) r.context.detach(r, REQUEST_TIMEOUT())
+    }, ms)
+  }
```

Comment:
define a static function

```
function onrequesttimeout(req) {
  if (req.context) ...
}
```

and then update the above to be req.timeout = setTimeout(onrequesttimeout, ms, req)